views/js/runner/plugins/controls/timer/timer.js

```diff
@@ -208,37 +208,42 @@ define([
              * Update the timers.
              * It will remove, let, add or update the current timers based on the current context.
              */
-            var updateTimers = function updateTimers(checkStorage){
+            var updateTimers = function updateTimers(checkStorage) {
+                return new Promise(function (resolve) {
+                    _.forEach(timerTypes, function(type){
```

Comment:
A Promise is resolved only once. Here everything will be considered as resolved at the 1st round of the loop. 
Instead do something like  this : 

```js
 var updateTimers = function updateTimers(checkStorage) {
     //add all promises in an array
     var updates = _.map(timerTypes, function(type) {
         return new Promise(function(resolve) {
             var timerConfig = getTimerConfig(type);
             if (currentTimers[type]) {
                 if (!timerConfig) {
                     removeTimer(type);
                 } else if (currentTimers[type].id() !== timerConfig.id) {
                     removeTimer(type);
                     addTimer(type, timerConfig);
                 }
                 return resolve();
             } else if (timerConfig) {
                 if (checkStorage) {
                     //check for the last value in the storage
                     self.storage.getItem(timerConfig.id).then(function(savedTime) {
                         if (_.isNumber(savedTime) && savedTime >= 0) {
                             timerConfig.remaining = savedTime;
                         }
                         addTimer(type, timerConfig);
                         resolve();
                     }).catch(function() {
                         //add the timer even if the storage doesn't work
                         addTimer(type, timerConfig);
                         resolve();
                     });
                 } else {
                     addTimer(type, timerConfig);
                     return resolve();
                 }
             }
             //haven't check all the cases but resolve MUST be called
            return resolve();
         });
     });

     //this create a promise that resolve once all promises are resolved (in parrallel)
     return Promise.all(updates);
 };
```
(it's untested but you have the big picture)